rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -684,7 +666,120 @@ public void onFailure(Throwable caught) {
+                for (Node node : host.getNodes().values()) {
+                    recalculateStatistics(node);
+                }
+            }
+        }
+    }
+
+    private void recalculateStatistics(Node node) {
+        switch (node.getNodeState()) {
+            case BUSY:
+                model.setNumBusy(model.getNumBusy() + 1);
+                break;
+            case CONFIGURING:
+                model.setNumConfiguring(model.getNumConfiguring() + 1);
+                break;
+            case DEPLOYING:
+                model.setNumDeploying(model.getNumDeploying() + 1);
+                break;
+            case DOWN:
+                model.setNumDown(model.getNumDown() + 1);
+                break;
+            case FREE:
+                model.setNumFree(model.getNumFree() + 1);
+                break;
+            case LOST:
+                model.setNumLost(model.getNumLost() + 1);
+                break;
+            case TO_BE_REMOVED:
+                model.setNumToBeRemoved(model.getNumToBeRemoved() + 1);
+                break;
+        }
+        if (node.isLocked()) {
+            model.setNumLocked(model.getNumLocked() + 1);
+        }
+    }
+
+    private NodeSource parseNodeSource(JSONObject nsObj) {
+        String sourceName = nsObj.get("sourceName").isString().stringValue();
+        String sourceDescription = getJsonStringNullable(nsObj, "sourceDescription");
+        String nodeSourceAdmin = nsObj.get("nodeSourceAdmin").isString().stringValue();
+        String nodeSourceStatus = getJsonStringNullable(nsObj, "nodeSourceStatus");
+        String eventType = getJsonStringNullable(nsObj, "eventType");
+        return new NodeSource(sourceName, sourceDescription, nodeSourceAdmin, nodeSourceStatus, eventType);
+    }
+
+    private Node parseNode(JSONObject nodeObj) {
+        String hostName = nodeObj.get("hostName").isString().stringValue();
+        String nss = nodeObj.get("nodeSource").isString().stringValue();
+
+        String eventType = getJsonStringNullable(nodeObj, "eventType");
+
+        String nodeUrl = nodeObj.get("nodeUrl").isString().stringValue();
+        String nodeState = nodeObj.get("nodeState").isString().stringValue();
+        String nodeInfo = nodeObj.get("nodeInfo").isString().stringValue();
+        String timeStampFormatted = nodeObj.get("timeStampFormatted").isString().stringValue();
+        long timeStamp = Math.round(nodeObj.get("timeStamp").isNumber().doubleValue());
+        String nodeProvider = nodeObj.get("nodeProvider").isString().stringValue();
+
+        String nodeOwner = getJsonStringNullable(nodeObj, "nodeOwner");
+        String vmName = getJsonStringNullable(nodeObj, "vmname");
+        String description = getJsonStringNullable(nodeObj, "nodeInfo");
+        String defaultJMXUrl = getJsonStringNullable(nodeObj, "defaultJMXUrl");
+        String proactiveJMXUrl = getJsonStringNullable(nodeObj, "proactiveJMXUrl");
+
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)